### PR TITLE
Improve dot products in `PGIsmallness.__call__` and update docstring

### PR DESCRIPTION
**Pull request by johnweis0480**
#### Summary
Change the `__call__` method in PGI smallness to avoid for loops through numpy arrays. Fix the docstring to reflect that physical properties can still be correlated if `approx_eval=True` and `approx_gradient=True`.

#### PR Checklist
* [x] If this is a work in progress PR, set as a Draft PR
* [x] Linted my code according to the [style guides](https://docs.simpeg.xyz/latest/content/getting_started/contributing/code-style.html).
* [x] Added [tests](https://docs.simpeg.xyz/latest/content/getting_started/contributing/testing.html) to verify changes to the code.
* [x] Added necessary documentation to any new functions/classes following the
      expect [style](https://docs.simpeg.xyz/latest/content/getting_started/contributing/documentation.html).
* [x] Marked as ready for review (if this is was a draft PR), and converted 
      to a Pull Request
* [x] Tagged ``@simpeg/simpeg-developers`` when ready for review.

### CODE REVIEW

File: simpeg/regularization/pgi.py [hunk 1]
```diff
@@ -70,10 +70,10 @@ class PGIsmallness(Smallness):
         ``tensor``, ``QuadTree`` or ``Octree`` meshes.
     approx_gradient : bool
         If ``True``, use the L2-approximation of the gradient by assuming
-        physical property values of different types are uncorrelated.
```
Comment (thibaut-kobold) [inline simpeg/regularization/pgi.py:73]: Oh yeah I remember correcting that statement somewhere else (maybe directives).
Disclaimer: I did not write those docstring. But the fault lies with me if there are inaccurate (I should have wrote them in the first place). That is to say, the person who wrote those did to their best of their understanding, but you should feel free to improve it.

**Review by santisoler — APPROVED** [review]
Looks great! Thanks @johnweis0480 for pushing this. I'm merging this!

~I'm not quite sure if the word distinct is the right replacement for uncorrelated in this case. I'd like to hear @thibaut-kobold's opinion on that change.~

**Comment by thibaut-kobold** [comment]
> Looks great! Thanks @johnweis0480 for pushing this.
> 
> I'm not quite sure if the word _distinct_ is the right replacement for _uncorrelated_ in this case. I'd like to hear @thibaut-kobold's opinion on that change.

I am fine with it.
"uncorrelated" was wrong (what does that even mean in this context?). "Distinct" is better description of when the L2 approximation is correct (when the physical properties distribution of each units are not overlapping).

**Comment by santisoler** [comment]
haha I commented that before seeing your review (sorry!). Then I removed that line from my comment because you already provided an answer. Thanks for that super swift reply haha.

**Comment by santisoler** [comment]
Restored and striked out!

**Pull request merged**